pkg/auth/ldaputil/query.go

```diff
@@ -239,6 +260,9 @@ func QueryForEntries(clientConfig *LDAPClientConfig, query *ldap.SearchRequest)
 	glog.V(4).Infof("searching LDAP server %v://%v at dn=%q with scope %v for %s requesting %v", clientConfig.Scheme, clientConfig.Host, query.BaseDN, query.Scope, query.Filter, query.Attributes)
 	searchResult, err := connection.Search(query)
 	if err != nil {
+		if strings.Contains(err.Error(), "LDAP Result Code 32") {
```

Comment:
Upstream is not typed. This occurs when the `baseDN` of a search refers to an object that doesn't exist in the tree.